libfreerdp/codec/h264_ffmpeg.c

```diff
@@ -240,22 +241,32 @@ static int libavcodec_decompress(H264_CONTEXT* h264, const BYTE* pSrcData, UINT3
 	do
 	{
 #ifdef WITH_VAAPI
-		status = avcodec_receive_frame(sys->codecDecoderContext,
-		                               sys->hwctx ? sys->hwVideoFrame : sys->videoFrame);
-#else
-		status = avcodec_receive_frame(sys->codecDecoderContext, sys->videoFrame);
+		if (h264->VaapiEnabled != 0)
```

Comment:
no need for this, if `VAAPI` was requested and found you have `sys->hwctx`

Reply:
Removedd